builtin/providers/azure/resource_azure_sql_database_server_firewall_rule.go

```diff
@@ -0,0 +1,222 @@
+package azure
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/management/sql"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceAzureSqlDatabaseServerFirewallRule returns the *schema.Resource
+// associated to a firewall rule of a database server in Azure.
+func resourceAzureSqlDatabaseServerFirewallRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzureSqlDatabaseServerFirewallRuleCreate,
+		Read:   resourceAzureSqlDatabaseServerFirewallRuleRead,
+		Update: resourceAzureSqlDatabaseServerFirewallRuleUpdate,
+		Delete: resourceAzureSqlDatabaseServerFirewallRuleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"database_server_names": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+			"start_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"end_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+// resourceAzureSqlDatabaseServerFirewallRuleCreate does all the necessary API
+// calls to create the SQL Database Server Firewall Rule on Azure.
+func resourceAzureSqlDatabaseServerFirewallRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	sqlClient := meta.(*Client).sqlClient
+
+	name := d.Get("name").(string)
+	params := sql.FirewallRuleCreateParams{
+		Name:           name,
+		StartIPAddress: d.Get("start_ip").(string),
+		EndIPAddress:   d.Get("end_ip").(string),
+	}
+
+	// loop over all the database servers and apply the firewall rule to each:
+	serverNames := d.Get("database_server_names").(*schema.Set).List()
+	for _, srv := range serverNames {
+		serverName := srv.(string)
+
+		log.Printf("[INFO] Sending Azure Database Server Firewall Rule %q creation request for Server %q.", name, serverName)
+		if err := sqlClient.CreateFirewallRule(serverName, params); err != nil {
+			return fmt.Errorf("Error creating Azure Database Server Firewall Rule %q for Server %q: %s", name, serverName, err)
+		}
+	}
+
+	d.SetId("name")
```

Comment:
This should be `SetId(name)`, otherwise every firewall rule will have the same ID. :grinning: 

Might want to start with a failing acceptance test here by making a test case with two rules - I'd expect that to fail without fixing this.